WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorDataProvider.swift

```diff
@@ -93,16 +95,20 @@ extension PaidDomainSuggestionViewModel {
 /// Provides domain suggestions that are paid.
 final class PaidDomainSelectorDataProvider: DomainSelectorDataProvider {
     private let stores: StoresManager
+    private let hasDomainCredit: Bool
 
-    init(stores: StoresManager = ServiceLocator.stores) {
+    init(stores: StoresManager = ServiceLocator.stores, hasDomainCredit: Bool) {
         self.stores = stores
+        self.hasDomainCredit = hasDomainCredit
     }
 
     @MainActor
     func loadDomainSuggestions(query: String) async throws -> [PaidDomainSuggestionViewModel] {
-        try await withCheckedThrowingContinuation { continuation in
+        let hasDomainCredit = hasDomainCredit
+        return try await withCheckedThrowingContinuation { continuation in
```

Comment:
Nit: 
Could we instead capture `hasDomaincredit` like the following? I believe that it will be clear that we use that inside the await block. 

```suggestion
        return try await withCheckedThrowingContinuation { [hasDomainCredit] continuation in
```

I hope there is no difference in the behaviour. Please correct me if I am wrong.

Reply:
this looks nicer 👍 updated in https://github.com/woocommerce/woocommerce-ios/pull/8771/commits/eb463a9e82af0a968fd89cd0f608069e2f42a4bc